teos/src/config.rs

```diff
@@ -225,13 +228,62 @@ impl Config {
                         _ => 8332,
                     }
                 }
-                Ok(())
+                network
             }
             Err(_) => {
-                Err(ConfigError(format!("btc_network not recognized. Expected {{bitcoin, testnet, signet, regtest}}, received {}",
-                self.btc_network)))
+                return Err(ConfigError(format!("btc_network not recognized. Expected {{bitcoin, testnet, signet, regtest}}, received {}",
+                self.btc_network)));
+            }
+        };
+
+        // Don't perform that sanity check in testing because we won't have a bitcoind instance running.
+        #[cfg(not(test))]
+        {
+            let real_btc_network = self.get_bitcoind_network().await?;
+            if real_btc_network != _teos_btc_network {
+                return Err(ConfigError(format!(
+                    "btc_network conflict. bitcoind is running on {} but teosd is set to run on {}",
+                    real_btc_network, _teos_btc_network
+                )));
             }
         }
+
+        Ok(())
+    }
+
+    /// Gets bitcoind's network.
+    pub async fn get_bitcoind_network(&self) -> Result<Network, ConfigError> {
+        // A wrapper class around `Network` to convert it from `JsonResponse`.
+        struct BtcNetwork(Network);
+        impl TryInto<BtcNetwork> for JsonResponse {
+            type Error = std::io::Error;
+            fn try_into(self) -> std::io::Result<BtcNetwork> {
+                Ok(BtcNetwork(Network::from_str(
+                    self.0["chain"].as_str().ok_or_else(|| {
+                        std::io::Error::new(
+                            std::io::ErrorKind::InvalidData,
+                            "'chain' option was not found",
+                        )
+                    })?,
+                )?))
+            }
+        }
+
+        // Get a bitcoind RPC client.
```

Comment:
You don't need to do this manually:

https://docs.rs/bitcoincore-rpc-json/0.16.0/bitcoincore_rpc_json/struct.GetBlockchainInfoResult.html

Reply:
Have in mind we are not really using the main implementation block from https://github.com/talaia-labs/rust-teos/blob/master/teos/src/bitcoin_cli.rs. For regular rpc, we are using https://docs.rs/bitcoincore-rpc/0.16.0/bitcoincore_rpc/struct.Client.html

Reply:
I actually kind of changed my mind here, see the most recent review. We can do some kind of hybrid version of this (using `bitcore-rpc` for the parsing) or go with this but under `bitcoin_cli.rs`